executor: limit the value of CPU shares (#19935)
hashicorp/nomad: .changelog/19935.txt

```release-note:bug

> client: Ensure the value for CPU weight is within the allowed range

Reviewer: ```suggestion
client: Ensure the value for CPU shares are within the allowed range
```
